diff --git a/node_modules/docker-run/index.js b/node_modules/docker-run/index.js
index ea7fc0d..d974bc0 100644
--- a/node_modules/docker-run/index.js
+++ b/node_modules/docker-run/index.js
@@ -14,7 +14,7 @@ var endsWith = function(str, suffix) {
 var run = function(image, opts) {
   if (!opts) opts = {}

-  var request = docker(opts.host, {version:'v1.14'})
+  var request = docker(opts.host, {version:'v1.45'})
   var that = new events.EventEmitter()
   var tty = !!opts.tty

