diff --git a/node_modules/prebuildify-cross/guest.js b/node_modules/prebuildify-cross/guest.js
index dfd9eb7..cfd4724 100644
--- a/node_modules/prebuildify-cross/guest.js
+++ b/node_modules/prebuildify-cross/guest.js
@@ -6,7 +6,7 @@ const cp = require('child_process')
 
 // TODO: fix permissions of WORKDIR in prebuild images
 const cwd = '/home/node/app'
-const files = JSON.parse(process.env.PREBUILDIFY_CROSS_FILES)
+const files = JSON.parse(fs.readFileSync('/input/.PREBUILDIFY_CROSS_FILES.json', 'utf-8'))
 const argv = process.argv.slice(2)
 
 // Copy host files to working directory
diff --git a/node_modules/prebuildify-cross/index.js b/node_modules/prebuildify-cross/index.js
index 0c37045..fac8eda 100644
--- a/node_modules/prebuildify-cross/index.js
+++ b/node_modules/prebuildify-cross/index.js
@@ -1,6 +1,7 @@
 'use strict'
 
 const packlist = require('npm-packlist')
+const fs = require('fs')
 const tar = require('tar-fs')
 const dockerPull = require('@vweevers/docker-pull')
 const dockerRun = require('docker-run')
@@ -25,6 +26,8 @@ module.exports = function (opts, callback) {
   const modules = opts.modules ? path.resolve(cwd, opts.modules) : null
 
   const files = JSON.stringify(packageFiles(cwd))
+  fs.writeFileSync('.PREBUILDIFY_CROSS_FILES.json', files)
+
   const prebuilds = path.join(cwd, 'prebuilds')
   const log = logger.create(process.stderr, { showCursor: true })
 
@@ -86,7 +89,6 @@ module.exports = function (opts, callback) {
       argv: ['-'].concat(argv),
       volumes,
       env: {
-        PREBUILDIFY_CROSS_FILES: files,
         // Disable npm update check
         NO_UPDATE_NOTIFIER: 'true'
       }
@@ -142,8 +144,8 @@ function prebuildifyArgv (argv, image) {
   }
 
   // TODO: move this to the docker images (https://github.com/prebuild/docker-images/issues/11)
-  if (/^(ghcr\.io\/)?prebuild\/(linux|android)-arm/.test(image)) argv.push('--tag-armv')
-  if (/^(ghcr\.io\/)?prebuild\/(centos|alpine)/.test(image)) argv.push('--tag-libc')
+  if (/\/(linux|android)-arm/.test(image)) argv.push('--tag-armv')
+  if (/\/(centos|alpine)/.test(image)) argv.push('--tag-libc')
 
   return argv
 }
